packages/api-server/api_server/rmf_io/test_rmf_io.py

```diff
@@ -255,3 +259,72 @@ def on_building_map(building_map: dict):
             self.static_files.add_file.call_args[0][1],
             "test_name/L1-test_image.thbyxgrllndgeciymb3a47hf2re5p7no.png",
         )
+
+
+class TestRmfIO_JWTAuth(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self):
+        self.client = socketio.AsyncClient()
+        self.rmf_gateway = RmfGateway()
+        self.sio = socketio.AsyncServer(async_mode="aiohttp")
+        self.static_files = MagicMock(StaticFilesRepository)
+        self.static_files.add_file.return_value = "/test_url"
+        self.test_cert = f"{os.path.dirname(__file__)}/test_data/test.pem"
+        self.auth = JwtAuthenticator(self.test_cert)
+        logger = logging.Logger("test")
+        logger.setLevel("CRITICAL")
+        self.rmf_io = RmfIO(
+            self.sio,
+            self.rmf_gateway,
+            self.static_files,
+            authenticator=self.auth,
+            logger=logger,
+        )
+        self.app = aiohttp.web.Application()
+        self.sio.attach(self.app)
+        self.runner = aiohttp.web.AppRunner(self.app)
+        await self.runner.setup()
+        self.site = aiohttp.web.TCPSite(self.runner, "127.0.0.1", 0)
+        await self.site.start()
+        self.server_port = self.runner.addresses[0][1]
+
+    async def asyncTearDown(self):
+        await self.client.disconnect()
+        await self.runner.cleanup()
+
+    async def test_fail_with_no_auth(self):
```

Comment:
Maybe I'm not getting it right, but aren't those tests always going to pass?

Reply:
turns out the tests are broken because the python client doesn't support auth, now using nodejs client via subprocess